internal/docstore/mongodocstore/mongo.go

```diff
@@ -253,6 +289,21 @@ func makeFilter(doc driver.Document) (filter bson.D, id, rev interface{}, err er
 	return filter, id, rev, nil
 }
 
+func (c *collection) docID(doc driver.Document) (interface{}, error) {
+	if c.idField != "" {
+		id, err := doc.GetField(c.idField)
+		if err != nil {
+			return nil, gcerr.New(gcerr.InvalidArgument, err, 2, "document missing ID")
+		}
+		return id, nil
+	}
+	id := c.idFunc(doc.Origin)
```

Comment:
Should it check `idFunc != nil`?

Reply:
That's not possible because of the logic on lines 72 and 73.